frontends/precinct-scanner/src/screens/poll_worker_screen.tsx

```diff
@@ -299,138 +277,233 @@ export function PollWorkerScreen({
     await printer.print({ sides: 'one-sided' });
   }
 
-  const [confirmOpenPolls, setConfirmOpenPolls] = useState(false);
-  const [
-    pollWorkerFlowState,
-    setPollWorkerFlowState,
-  ] = useState<PollWorkerFlowState>(PollWorkerFlowState.INIT);
-  function openConfirmOpenPollsModal() {
-    return setConfirmOpenPolls(true);
-  }
-  function closeConfirmOpenPollsModal() {
-    return setConfirmOpenPolls(false);
+  const [pollWorkerFlowState, setPollWorkerFlowState] = useState<
+    PollWorkerFlowState | undefined
+  >(
+    isPollsOpen
+      ? PollWorkerFlowState.CLOSE_POLLS_FLOW__CONFIRM
+      : PollWorkerFlowState.OPEN_POLLS_FLOW__CONFIRM
+  );
+  function showAllPollWorkerActions() {
+    return setPollWorkerFlowState(undefined);
   }
-  async function openPollsAndHandleZeroReport() {
-    setIsHandlingTallyReport(true);
+
+  async function openPolls() {
+    setPollWorkerFlowState(PollWorkerFlowState.OPEN_POLLS_FLOW__PROCESSING);
     if (hasPrinterAttached) {
       await printTallyReport();
     } else {
       await saveTally();
     }
     togglePollsOpen();
-    setIsHandlingTallyReport(false);
-    setPollWorkerFlowState(PollWorkerFlowState.POLLS_OPENED_COMPLETE);
-    closeConfirmOpenPollsModal();
+    setPollWorkerFlowState(PollWorkerFlowState.OPEN_POLLS_FLOW__COMPLETE);
   }
 
-  const [confirmClosePolls, setConfirmClosePolls] = useState(false);
-  function openConfirmClosePollsModal() {
-    return setConfirmClosePolls(true);
-  }
-  function closeConfirmClosePollsModal() {
-    return setConfirmClosePolls(false);
-  }
-  async function closePollsAndHandleTabulationReport() {
-    setIsHandlingTallyReport(true);
+  async function closePolls() {
+    assert(electionDefinition);
+    assert(currentUserSession);
+    setPollWorkerFlowState(PollWorkerFlowState.CLOSE_POLLS_FLOW__PROCESSING);
     if (hasPrinterAttached) {
       await printTallyReport();
     } else {
       await saveTally();
     }
+    if (scannedBallotCount > 0) {
+      await saveCvrExportToUsb({
+        electionDefinition,
+        machineConfig,
+        scannedBallotCount,
+        isTestMode: !isLiveMode,
+        openFilePickerDialog: false,
+      });
+    }
+    await sleep(1000);
```

Comment:
@benadida if I didn't add this I got an error about the usb being busy when I tried to eject it, do you think this is the right amount of time to wait?

Reply:
Uggh, this asynchronicity without clear end is painful. I think we need to fix this for good for m14. For now, I would urge us to be particularly conservative and do 4s.

Reply:
Related: https://github.com/votingworks/vxsuite/issues/1583